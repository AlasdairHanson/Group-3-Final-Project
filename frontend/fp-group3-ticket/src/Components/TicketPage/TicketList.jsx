import React from "react";
import { Accordion } from "react-bootstrap";
import Ticket from "./Ticket";
const TicketList = ({ data, error, isLoaded }) => {
  if (!isLoaded) {
    return <p> Please wait.... we are loading your information</p>;
  } else {
    return (
      <>
        <Accordion defaultActiveKey="0">
          {data.map((ticket) => (
            <Ticket
              key={ticket.id}
              obj={ticket}
              id={ticket.id}
              acc_id={ticket.id}
              title={ticket.title}
              topic={ticket.topic}
              desc={ticket.content}
              time={ticket.time}
              status={ticket.status}
              trainee={ticket.author}
              trainer={ticket.trainer}
              priority={ticket.priority}
              cohort={ticket.cohort}
            />
          ))}
        </Accordion>
      </>
    );
  }
};
export default TicketList;
