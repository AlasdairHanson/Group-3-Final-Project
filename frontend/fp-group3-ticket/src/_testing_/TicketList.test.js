import { render, screen } from '@testing-library/react';
import TicketList from '../Components/TicketPage/TicketList';
import '@testing-library/jest-dom';
import {create} from 'react-test-renderer';

jest.mock(`../Components/TicketPage/Ticket`, () => {
  return function TicketMock(props){
    return(<p>Mocked Ticket</p>)
  }
});

describe(`Snapshot test of TicketList`, () => {

  test(` TicketList should match the snapshot every time it is rendered`, () => {
    const instance = create(<TicketList/>).toJSON(); 
    expect(instance).toMatchSnapshot();
  });
});