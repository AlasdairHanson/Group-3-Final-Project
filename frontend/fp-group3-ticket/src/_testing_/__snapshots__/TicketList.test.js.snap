// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of TicketList  TicketList should match the snapshot every time it is rendered 1`] = `
<p>
   Please wait.... we are loading your information
</p>
`;
